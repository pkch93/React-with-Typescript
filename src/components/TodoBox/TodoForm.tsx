import React from 'react';

import styled from 'styled-components';

const Wrapper = styled.div`
    display: flex;
    justify-content: space-between;
`;

const Input = styled.input`
    width: 75%;
    height: 30px;
    margin-left: 2%;
`;

const Button = styled.button<{ color: string }>`
    background-color: ${({ color }) => color};
    border: none;
    width: 20%;
    margin-right: 2%;
    font-size: 20px;
`;

interface TodoFormProps {
    color: string;
}

const TodoForm: React.FC<TodoFormProps> = ({ color }) => (
    <Wrapper>
        <Input type="text" placeholder="할 일을 입력하세요!"/>
        <Button color={color} type="submit">Todo</Button>
    </Wrapper>
);

export default TodoForm;