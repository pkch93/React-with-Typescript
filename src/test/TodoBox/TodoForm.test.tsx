import React from 'react';
import { render } from '@testing-library/react';

import TodoForm from '../../components/TodoBox/TodoForm';

describe('<TodoForm />', () => {
    it("matches snapshot", () => {
        const utils = render(<TodoForm />);
        expect(utils.container).toMatchSnapshot();
    });

    it('have one submit button and one input box', () => {
        const target = render(<TodoForm />);
        target.getByPlaceholderText("할 일을 입력하세요!");
        target.getByText("Todo");
    });
});