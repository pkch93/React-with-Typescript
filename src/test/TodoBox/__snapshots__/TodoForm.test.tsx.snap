// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoForm /> matches snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa ivgsgg"
  >
    <input
      class="sc-bwzfXH eEIFdd"
      placeholder="할 일을 입력하세요!"
      type="text"
    />
    <button
      class="sc-htpNat eOtgLR"
      color="#66d9e8"
      type="submit"
    >
      Todo
    </button>
  </div>
</div>
`;
